app/registration.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Register components (via a list of glob patterns)
+ *
+ * Copyright © 2015 Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NonComposerComponentRegistration;
+
+use RuntimeException;
+
+/**
+ * Include files from a list of glob patterns
+ *
+ * @throws RuntimeException
+ * @return void
+ */
+function main()
+{
+    $globPatterns = require __DIR__ . '/etc/registration_globlist.php';
+    $baseDir = __DIR__ . '/';
+
+    foreach ($globPatterns as $globPattern) {
+        // Sorting is disabled intentionally for performance improvement
+        $files = glob($baseDir . $globPattern, GLOB_NOSORT);
+        if ($files === false) {
+            throw new RuntimeException("glob(): error with '$baseDir$globPattern'");
+        }
+        array_map(__NAMESPACE__ . '\file', $files);
```

Comment:
I would suggest changing this line to simple this. It accomplishes the objective of variable scope isolation, as well as eliminates the string reference to a callback function.
```php
array_map(function ($file) { require_once $file; }, $files);
```